### PR TITLE
clean up non-interacting periodic and non-conformal alg

### PR DESCRIPTION
**Pull request by spdomin**
* dof STATUS is still a mess. Very fragile and in great need of a
  redisn. This push at least allows for non-interacting dg and periodic
  bcs to be properly handled... The previous implementation was confusing
  the dof status for a non conformal scheme when that node was not a
  perioidc node, however, periodicity was activated.

Zero diffs; several use cases testng this all work now.

### CODE REVIEW

File: src/TpetraLinearSystem.C [hunk 1]
```diff
@@ -2218,33 +2220,24 @@ int getDofStatus_impl(stk::mesh::Entity node, const Realm& realm)
       return DS_SharedNotOwnedDOF;
```
Comment (alanw0) [inline src/TpetraLinearSystem.C]: I'm confused by "if owned return shared". Looks like this already existed before your commit.

File: src/TpetraLinearSystem.C [hunk 2]
```diff
@@ -2271,7 +2264,16 @@ int getDofStatus_impl(stk::mesh::Entity node, const Realm& realm)
       }
     }
   }
-
+  
+  // finally, if non conformal is active
+  if ( nonConformalActive ) {
+    if (entityIsOwned)
+      return DS_OwnedDOF;
+    if (!entityIsOwned && (entityIsGhosted || entityIsShared)){
```
Comment (alanw0) [inline src/TpetraLinearSystem.C]: I'm still not sure this is correct either...

Comment (spdomin) [reply src/TpetraLinearSystem.C]: This is why dofStatus is a mess. In STK, we have clean definitions of owned, shared and ghosted.  However, dofStatus does not respect this. For the non-conformal all to work properly, a ghosted node due to the non-conformal all needs to be listed as "sharedNotOwned". Every part of the code in Tpetra assumes this is the case.

Sadly, in that symmetry bc example, the ghosted node due to non-conformal was "correct" in listing it as "shared not owned".

I guess that in dofStatus, we have taken over the definition of shared. It means not owned, however, ghosted or shared when the non-conformal all is activated.

**Comment by spdomin** [comment]
agreed... This is just a bandage for this periodic/nc-alg use case.

**Pull request merged**